robot-name/Pratster95/robot_name.go

```diff
@@ -0,0 +1,40 @@
+package robotname
+
+import (
+	"fmt"
+	"math/rand"
+)
+
+// Define the Robot type here.
+type Robot struct {
+	name string
+}
+
+var nameUsed = map[string]bool{}
+
+const max = 26 * 26 * 10 * 10 * 10
+
+func (r *Robot) Name() (string, error) {
+	if r.name != "" {
```

Comment:
I'm all in with proper names. I'd call it `robot` instead of `r`. It becomes easier to read the code and the compiler don't actually care about how many characters we use.

Reply:
**TLDR:** Forget my comment above.

It was brought to my attention that this could not be the best approach.

There's is no convention regarding this (that I could find) but the most common thing to find is one letter names like you used. Searching a bit more about this, I found out a wiki page in the go repository that collects common review comments. [It says this about receivers' names](https://github.com/golang/go/wiki/CodeReviewComments#receiver-names):

> The name of a method's receiver should be a reflection of its identity; often a one or two letter abbreviation of its type suffices (such as "c" or "cl" for "Client"). Don't use generic names such as "me", "this" or "self", identifiers typical of object-oriented languages that gives the method a special meaning. In Go, the receiver of a method is just another parameter and therefore, should be named accordingly. The name need not be as descriptive as that of a method argument, as its role is obvious and serves no documentary purpose. It can be very short as it will appear on almost every line of every method of the type; familiarity admits brevity. Be consistent, too: if you call the receiver "c" in one method, don't call it "cl" in another.

In short:

- One or two letters are more common for receivers' names.
- You should keep consistency.
- You can use whatever name you want but as it's role is well defined, you don't need to be descriptive.

Discussing about this, the only corner case which I could find is having more than one struct with receivers in the same go file. In those cases, the naming would become more confusing. Of course, declaring the receivers just after the struct would improve the readability.